core.go

```diff
@@ -34,6 +58,9 @@ func (c *core) With(fs []zapcore.Field) zapcore.Core {
 }
 
 func (c *core) Check(ent zapcore.Entry, ce *zapcore.CheckedEntry) *zapcore.CheckedEntry {
+	if c.cfg.EnableBreadcrumbs && c.cfg.BreadcrumbLevel.Enabled(ent.Level) {
```

Comment:
This seems redundant. You should drop this and use the new `LevelEnabler` you created, or use it like this and drop the new `LevelEnabler` (that's what I would do personally).

One small detail: I'd move the check for breadcrumbs under the check for the main Level, as it's more of a "fallback". But, really, just a detail...